starter/src/main/java/com/example/starter/base/impl/MainView.java

```diff
@@ -15,15 +19,29 @@
 @Route(value = "", layout = MainLayout.class)
 public class MainView extends VerticalLayout {
 
+    public GreetService getGreetService() throws Exception {
+        BundleContext bundleContext = FrameworkUtil.getBundle(MainView.class).getBundleContext();
+        ServiceReference<GreetService> serviceReference = bundleContext.getServiceReference(GreetService.class);
+        if( serviceReference == null) {
+            throw new Exception("Greet service not currently available, install it into Karaf to get proper greeting");
+        }
+        return bundleContext.getService(serviceReference);
+    }
+
     public MainView() {
 
         // Use TextField for standard text input
         TextField textField = new TextField("Your name");
         textField.addThemeName("bordered");
         // Button click listeners can be defined as lambda expressions
-        GreetService greetService = new GreetService();
-        Button button = new Button("Say hello", e -> Notification
-                .show(greetService.greet(textField.getValue())));
+        Button button = new Button("Say hello", e -> {
+            try {
+                Notification.show(getGreetService().greet(textField.getValue()));
+            } catch (Exception exception) {
```

Comment:
Catching `Exception` is a very bad practice overall .
Here it's not needed because greet service should present always as mentioned above.

Reply:
Refactored to use the optional service in a method that returns the greeting string directly.